packages/engine/Source/Shaders/Builtin/Functions/applyHSBShift.glsl

```diff
@@ -0,0 +1,21 @@
+/**
+ * Apply a HSB color shift to an RGB color.
+ *
+ * @param {vec3} rgb The color in RGB space.
+ * @param {vec3} hsbShift The amount to shift each component. The xyz components correspond to hue, saturation, and brightness. Shifting the hue by +/- 1.0 corresponds to shifting the hue by a full cycle. Saturation and brightness are clamped between 0 and 1 after the adjustment
+ *
+ * @return {vec3} The RGB color after shifting in HSB space and clamping saturation and brightness to a valid range.
+ */
+vec3 czm_applyHSBShift(vec3 rgb, vec3 hsbShift) {
+    // Convert rgb color to hsb
+    vec3 hsb = czm_RGBToHSB(rgb);
+
+    // Perform hsb shift
+    // Hue cycles around so no clamp is needed.
+    hsb.x += hsbShift.x; // hue
+    hsb.y = clamp(hsb.y + hsbShift.y, 0.0, 1.0); // saturation
+    hsb.z = clamp(hsb.z + hsbShift.z, 0.0, 1.0); // brightness
```

Comment:
ohh that's what that ternary operator was for in the original function. I'll update the new function.

Reply:
I added the old code, but with a parameter `bool ignoreBlackPixels` so if some future shader wanted a HSB shift but did not want that behavior, it can be turned off.